test/integration/user/test/featureInViewport.test.js

```diff
@@ -0,0 +1,47 @@
+import * as carto from '../../../../src';
+import features from './features';
+import mapboxgl from '@carto/mapbox-gl';
+
+describe('viewport features', () => {
+    describe('given a list of features', () => {
+        describe('when the viewport is in a predefined place (center/zoom)', () => {
+            let map, source, viz, layer, div;
+            beforeEach(() => {
+                div = document.createElement('div');
+                div.id = 'map';
+                div.style.width = '500px';
+                div.style.height = '500px';
+                div.style.position = 'absolute';
+                document.body.style.margin = '0';
+                document.body.style.padding = '0';
+                document.body.appendChild(div);
+
+                map = new mapboxgl.Map({
+                    container: 'map',
+                    style: 'https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json',
+                    center: [-6.754536880813703, 35.30378023840868],
```

Comment:
How can I check that this test is actually correct? Center, zoom and feature geometries seem magic

Reply:
I created a dataset by hand with several squares. Given the map size in pixels the center and the zoom some of them are inside/outside/colliding with viewport so all cases are covered

suggestions to improve this are welcome :)

Reply:
> How can I check that this test is actually correct?

open the `.html` file with the browser

Reply:
Which `.hmtl`?

Reply:
Regarding the suggestions, I would write a render test instead of a user test. Render tests are standalone and you can actually pan and zoom-in and zoom-out (if you use a Mapbox GL map instead of the simple standalone Map integrator).

Reply:
strong disagree, render tests are intended to test rendering!

We are testing an expression loading a real scenario and [actually testing](https://github.com/CartoDB/carto-vl/pull/635/files#diff-c07b14a7498127fa2e3a3d9b3064a1d1R37) the value of the expression without dirty hacks. This is faster, more precise, more stable and doesn't get affected by browser-rendering issues.

Reply:
> Which .hmtl?

1. Change [this line](https://github.com/CartoDB/carto-vl/pull/635/files#diff-c07b14a7498127fa2e3a3d9b3064a1d1R5) and put an `fdescribe` to run only those tests
2. Coment the `afterEach` block to prevent the map to be removed from the browser
3. Run tests in debug mode with: `test:user:watchc``
4. Youll see a karma browser, click on `debug` and you'll see the test case:

![screen shot 2018-07-13 at 10 54 08](https://user-images.githubusercontent.com/2657897/42682628-73e72c40-868b-11e8-8454-d35dc7295ba1.png)